internal/api/front/perses/src/app/project/dashboard/dashboard.service.ts

```diff
@@ -13,17 +13,18 @@
 
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
-import { ErrorHandlingService } from '../../shared/service/error-handling.service';
 import { Observable } from 'rxjs';
+import { DashboardModel } from './dashboard.model';
 import { UrlBuilderUtil } from '../../shared/utils/url-builder.util';
+import { ErrorHandlingService } from '../../shared/service/error-handling.service';
 import { catchError } from 'rxjs/operators';
-import { DashboardModel } from './dashboard.model';
 
 @Injectable({
   providedIn: 'root'
 })
 export class DashboardService {
   private readonly resource = 'dashboards';
+  currentDashboard: DashboardModel = {} as DashboardModel;
```

Comment:
I'm not really sure it's the right way to do it. It's maybe fine for the moment, but the following scenario would raise the issue:

1. I'm on the dashboard list, I'm selecting let's `dashboard A`.
2. I'm going to display the `dashboard A`. So far so good
3. I'm moving to the PrometheusRule
4. Imagine you have a link in the rule to see the `Dashboard B`. I'm clicking on this link.
5. I'm now back to the page to display a dashboard details. But actually instead of displaying `Dashboard B` it is displaying `Dashboard A` because it is what it is cached in the `DashboardService`

Reply:
I didn't have a good idea right now, maybe a `BehaviorSubject` would make it. @celian-garcia WDYT

Reply:
Actually it will work, because of the second condition of this `if` in dashboard-details : 
```typescript
        if (Object.keys(this.service.currentDashboard).length !== 0
        && this.service.currentDashboard.metadata.name === this.name) {
          this.dashboard = this.service.currentDashboard;
          this.feedDashboard();
        } else {
          this.getDashboard();
        }
```

But yes it's not perfect, in this case a deprecated variable will still continue to live within the service until you go back to dashboardList (so actually the concept of "current dashboard" is wrong)

I told a bit more about this here : https://github.com/perses/perses/pull/41#discussion_r640153924

If I got it right, since -list and -details are sibblings components an other approach would be to have -list sending data to the parent and then the parent transmitting it to -details. But I didnt understand how to achieve this so far.

Reply:
mechanism with cache at service level + a guard has been implemented to replace this